example/game/src/main/java/no/nordicsemi/android/ble/example/game/client/view/ClientScreen.kt

```diff
@@ -0,0 +1,136 @@
+import no.nordicsemi.android.ble.ktx.state.ConnectionState
+import no.nordicsemi.android.common.navigation.NavigationManager
+import no.nordicsemi.android.common.permission.RequireBluetooth
+import no.nordicsemi.android.common.theme.view.NordicAppBar
+
+@Composable
+fun ClientScreen(
+    navigationManager: NavigationManager,
+) {
+    Column {
+        NordicAppBar(
+            text = stringResource(id = R.string.client),
+            onNavigationButtonClick = {
+                navigationManager.navigateUp()
+            }
+        )
+
+        RequireBluetooth {
+            val clientViewModel: ClientViewModel = hiltViewModel()
+            val state by clientViewModel.state.collectAsState()
+
+            when (state) {
+                ConnectionState.Initializing -> {
+                    OutlinedCard(
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .padding(32.dp),
+                    ) {
+                        Text(
+                            text = stringResource(id = R.string.initializing),
+                            modifier = Modifier.padding(16.dp)
+                        )
+                    }
+                }
+                ConnectionState.Connecting -> {
+                    OutlinedCard(
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .padding(32.dp),
+                    ) {
+                        Text(
+                            text = stringResource(id = R.string.scanning_for_server),
+                            modifier = Modifier.padding(16.dp)
+                        )
+                    }
+                }
+                ConnectionState.Ready -> {
+                    val question by clientViewModel.question.collectAsState()
+
+                    question?.let { q ->
+                        val correctAnswerId by clientViewModel.answer.collectAsState()
+                        val selectedAnswerId by clientViewModel.selectedAnswer
+                        val ticks by clientViewModel.ticks.collectAsState()
+                        val finalResult by clientViewModel.finalResult.collectAsState()
+
+                        if (finalResult?.isGameOver == true) ShowResult(result = finalResult!!.finalResult)
+                        else {
+                            QuestionScreenClient(
+                                question = q,
+                                correctAnswerId = correctAnswerId,
+                                selectedAnswerId = selectedAnswerId,
+                                ticks = ticks,
+                                modifier = Modifier.fillMaxWidth()
+                            ) { answerChosen ->
+                                clientViewModel.sendAnswer(answerChosen)
+                            }
+                        }
+                    } ?: run {
```

Comment:
I remember discussion with my dearest friend Roshan that Elvis operator works only for assignments (val a = b ?: c). Can you double check that what you wrote works?